feat(interview-prep): Converting and Tranfering Rosetta challenges

NitronR opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #XXXXX

<!-- Feel free to add any additional description of changes below this line -->

NitronR committed:
feat(interview-prep): Converting and Tranfering Rosetta challenges

NitronR commented:
@scissorsneedfoodtoo This is the next set of problems, can you please review this? I will try to add guide pages for these challenges as @moT01 suggested.

moT01 reviewed (changes requested):
That's very strange, when and where did `S-Expressions` get deleted from the meta? It helps if you can give a little explanation in the description when you find something like that @NitronR. I didn't review the challenges themselves yet.

The challenge files should all be changed to end with `.english.md` and not just `.md`

I created guide articles for these...
https://www.freecodecamp.org/forum/t/freecodecamp-challenge-guide-search-a-list-of-records/385315
https://www.freecodecamp.org/forum/t/freecodecamp-challenge-guide-self-referential-sequence/385317
https://www.freecodecamp.org/forum/t/freecodecamp-challenge-guide-semiprime/385318
https://www.freecodecamp.org/forum/t/freecodecamp-challenge-guide-set-consolidation/385319
https://www.freecodecamp.org/forum/t/freecodecamp-challenge-guide-set-of-real-numbers/385322
https://www.freecodecamp.org/forum/t/freecodecamp-challenge-guide-sha-1/385326
https://www.freecodecamp.org/forum/t/freecodecamp-challenge-guide-sha-256/385328

You just need to add the forumTopicId in the frontmatter.

**Note to self: add these solutions to the guide articles when this gets merged.**

NitronR commented:
@moT01 Thank you for adding the guide pages. I have added  the forumTopicIds and renamed the files. 
I corrected the position of `S-Expressions` so that it can be consistent with [Rosetta Code ordering](http://www.rosettacode.org/wiki/Category:Programming_Tasks). I am sorry, I forgot to mention it in the description.

moT01 commented:
Thanks @NitronR, 

> I corrected the position of S-Expressions

Ahh, I guess I missed the removal of it in the diff. I was looking through the file history and couldn't find where it got deleted.

scissorsneedfoodtoo commented on curriculum/challenges/english/08-coding-interview-prep/rosetta-code/search-a-list-of-records.english.md:
````diff
@@ -0,0 +1,95 @@
+<div id='js-seed'>
+
+```js
+var lst = [
````
const lst = [

scissorsneedfoodtoo commented on curriculum/challenges/english/08-coding-interview-prep/rosetta-code/search-a-list-of-records.english.md:
```diff
@@ -0,0 +1,95 @@
+  { name: 'Alexandria', population: 4.58 },
+  { name: 'Abidjan', population: 4.4 },
+  { name: 'Casablanca', population: 3.98 }
+];
```
Not a big deal, but we might consider moving `lst` below `searchCity` so the function that campers will work on is at the top of the editor.

scissorsneedfoodtoo commented on curriculum/challenges/english/08-coding-interview-prep/rosetta-code/search-a-list-of-records.english.md:
````diff
@@ -0,0 +1,95 @@
+<section id='solution'>
+
+```js
+var lst = [
+  { name: 'Lagos', population: 21.0 },
+  { name: 'Cairo', population: 15.2 },
+  { name: 'Kinshasa-Brazzaville', population: 11.3 },
+  { name: 'Greater Johannesburg', population: 7.55 },
+  { name: 'Mogadishu', population: 5.85 },
+  { name: 'Khartoum-Omdurman', population: 4.98 },
+  { name: 'Dar Es Salaam', population: 4.7 },
+  { name: 'Alexandria', population: 4.58 },
+  { name: 'Abidjan', population: 4.4 },
+  { name: 'Casablanca', population: 3.98 }
+];
+function searchCity(name) {
+  for (var i = 0; i < lst.length; i++) {
+    if (lst[i].name == name) return i;
+  }
+}
````
Not the biggest deal since users won't see this, but we could consider updating this solution to use ES6.

scissorsneedfoodtoo commented on curriculum/challenges/english/08-coding-interview-prep/rosetta-code/self-referential-sequence.english.md:
```diff
@@ -0,0 +1,144 @@
+
+## Description
+
+<section id='description'>
+There are several ways to generate a self-referential sequence. One very common one (the <a href="http://rosettacode.org/wiki/Look-and-say sequence" target="_blank">Look-and-say sequence</a>) is to start with a positive integer, then generate the next term by concatenating enumerated groups of adjacent alike digits:
+0, 10, 1110, 3110, 132110, 1113122110, 311311222110 ...
+The terms generated grow in length geometrically and never converge.
+Another way to generate a self-referential sequence is to summarize the previous term.
+Count how many of each alike digit there is, then concatenate the sum and digit for each of the sorted enumerated digits. Note that the first five terms are the same as for the previous sequence.
+0, 10, 1110, 3110, 132110, 13123110, 23124110 ... see <a href="http://rosettacode.org/wiki/oeis:A036058" target="_blank">The On-Line Encyclopedia of Integer Sequences</a>
+Sort the digits largest to smallest. Do not include counts of digits that do not appear in the previous term.
+Depending on the seed value, series generated this way always either converge to a stable value or to a short cyclical pattern. (For our purposes, I'll use converge to mean an element matches a previously seen element.) The sequence shown, with a seed value of 0, converges to a stable value of 1433223110 after 11 iterations. The seed value that converges most quickly is 22. It goes stable after t ...
+</section>
```
No code blocks here, but the spacing can match the original problem. Let's also remove one unnecessary link and go to the secure Rosetta Code page:

```
<section id='description'>
There are several ways to generate a self-referential sequence. One very common one (the <a href="https://rosettacode.org/wiki/Look-and-say sequence" target="_blank">Look-and-say sequence</a>) is to start with a positive integer, then generate the next term by concatenating enumerated groups of adjacent alike digits:

<pre>0, 10, 1110, 3110, 132110, 1113122110, 311311222110 ...</pre>

The terms generated grow in length geometrically and never converge.

Another way to generate a self-referential sequence is to summarize the previous term.

Count how many of each alike digit there is, then concatenate the sum and digit for each of the sorted enumerated digits. Note that the first five terms are the same as for the previous sequence.

<pre>0, 10, 1110, 3110, 132110, 13123110, 23124110 ...</pre>

Sort the digits largest to smallest. Do not include counts of digits that do not appear in the previous term.

Depending on the seed value, series generated this way always either converge to a stable value or to a short cyclical pattern. (For our purposes, I'll use converge to mean an element matches a previously seen element.) The sequence shown, with a seed value of 0, converges to a stable value of 1433223110 after 11 iterations. The seed value that converges most quickly is 22. It goes stable after the first element. (The next element is 22, which has been seen before.)
</section>
```

scissorsneedfoodtoo commented on curriculum/challenges/english/08-coding-interview-prep/rosetta-code/semiprime.english.md:
```diff
@@ -0,0 +1,116 @@
+## Instructions
+
+<section id='instructions'>
+Write a function determining whether a given number is semiprime.
```
Write a function that returns true if a number is semiprime, or false if it is not.

scissorsneedfoodtoo commented on curriculum/challenges/english/08-coding-interview-prep/rosetta-code/semiprime.english.md:
```diff
@@ -0,0 +1,116 @@
+## Description
+
+<section id='description'>
+Semiprime numbers are natural numbers that are products of exactly two (possibly equal) <a href="http://rosettacode.org/wiki/prime_number" target="_blank">prime numbers</a>.
+ 1679  =  23 &times; 73
```
````
Semiprime numbers are natural numbers that are products of exactly two (possibly equal) <a href="https://rosettacode.org/wiki/prime_number" target="_blank">prime numbers</a>.
<pre>1679  =  23 x 73</pre>
````

scissorsneedfoodtoo commented on curriculum/challenges/english/08-coding-interview-prep/rosetta-code/set-consolidation.english.md:
```diff
@@ -0,0 +1,165 @@
+The consolidation of the five sets:
+:<code>{H,I,K}</code>, <code>{A,B}</code>, <code>{C,D}</code>, <code>{D,B}</code>, and <code>{F,G,H}</code>
+Is the two sets:
+:<code>{A, C, B, D}</code>, and <code>{G, F, I, H, K}</code>
```
````
<section id='description'>
Given two sets of items then if any item is common to any set then the result of applying <i>consolidation</i> to those sets is a set of sets whose contents is:

<ul>
  <li>The two input sets if no common item exists between the two input sets of items.</li>
  <li>The single set that is the union of the two input sets if they share a common item.</li>
</ul>

Given N sets of items where N > 2 then the result is the same as repeatedly replacing all combinations of two sets by their consolidation until no further consolidation between set pairs is possible. If N < 2 then consolidation has no strict meaning and the input can be returned.

Here are some examples:

<strong>Example 1:</strong>

<span style="margin-left: 2em;">Given the two sets `{A,B}` and `{C,D}` then there is no common element between the sets and the result is the same as the input.</span>

<strong>Example 2:</strong>

<span style="margin-left: 2em;">Given the two sets `{A,B}` and `{B,D}` then there is a common element `B` between the sets and the result is the single set `{B,D,A}`.  (Note that order of items in a set is immaterial: `{A,B,D}` is the same as `{B,D,A}` and `{D,A,B}`, etc).</span>

<strong>Example 3:</strong>

<span style="margin-left: 2em;">Given the three sets `{A,B}` and `{C,D}` and `{D,B}` then there is no common element between the sets `{A,B}` and `{C,D}` but the sets `{A,B}` and `{D,B}` do share a common element that consolidates to produce the result `{B,D,A}`. On examining this result with the remaining set, `{C,D}`, they share a common element and so consolidate to the final output of the single set `{A,B,C,D}`</span>

<strong>Example 4:</strong>

<span style="margin-left: 2em;">The consolidation of the five sets:</span>

<span style="margin-left: 4em;">`{H,I,K}`, `{A,B}`, `{C,D}`, `{D,B}`, and `{F,G,H}`</span>

<span style="margin-left: 2em;">Is the two sets:</span>

<span style="margin-left: 4em;">`{A, C, B, D}`, and `{G, F, I, H, K}`</span>
</section>
````

scissorsneedfoodtoo commented on curriculum/challenges/english/08-coding-interview-prep/rosetta-code/set-of-real-numbers.english.md:
```diff
@@ -0,0 +1,161 @@
+
+## Description
+
+<section id='description'>
+All real numbers form the uncountable set R.  Among its subsets, relatively simple are the convex sets, each expressed as a range between two real numbers <i>a</i> and <i>b</i> where <i>a</i> ≤ <i>b</i>.  There are actually four cases for the meaning of "between", depending on open or closed boundary:
+<ul>
+  <li>[<i>a</i>, <i>b</i>]: {<i>x</i> | <i>a</i> ≤ <i>x</i> and <i>x</i> ≤ <i>b</i> }</li>
+  <li>(<i>a</i>, <i>b</i>): {<i>x</i> | <i>a</i> < <i>x</i> and <i>x</i> < <i>b</i> }</li>
+  <li>[<i>a</i>, <i>b</i>): {<i>x</i> | <i>a</i> ≤ <i>x</i> and <i>x</i> < <i>b</i> }</li>
+  <li>(<i>a</i>, <i>b</i>]: {<i>x</i> | <i>a</i> < <i>x</i> and <i>x</i> ≤ <i>b</i> }</li>
+</ul>
+Note that if <i>a</i> = <i>b</i>, of the four only [<i>a</i>, <i>a</i>] would be non-empty.
+<b>Task</b>
+<ul>
+  <li>Devise a way to represent any set of real numbers, for the definition of 'any' in the implementation notes below.</li>
+  <li>Provide methods for these common set operations (<i>x</i> is a real number; <i>A</i> and <i>B</i> are sets):</li>
+</ul>
+<i>x</i> ∈ <i>A</i>: determine if <i>x</i> is an element of <i>A</i>
+example: 1 is in [1, 2), while 2, 3, ... are not.
+<i>A</i> ∪ <i>B</i>: union of <i>A</i> and <i>B</i>, i.e. {<i>x</i> | <i>x</i> ∈ <i>A</i> or <i>x</i> ∈ <i>B</i>}
+example: [0, 2) ∪ (1, 3) = [0, 3); [0, 1) ∪ (2, 3] = well, [0, 1) ∪ (2, 3]
+<i>A</i> ∩ <i>B</i>: intersection of <i>A</i> and <i>B</i>, i.e. {<i>x</i> | <i>x</i> ∈ <i>A</i> and <i>x</i> ∈ <i>B</i>}
+example: [0, 2) ∩ (1, 3) = (1, 2); [0, 1) ∩ (2, 3] = empty set
+<i>A</i> - <i>B</i>: difference between <i>A</i> and <i>B</i>, also written as <i>A</i> \ <i>B</i>, i.e. {<i>x</i> | <i>x</i> ∈ <i>A</i> and <i>x</i> ∉ <i>B</i>}
+example: [0, 2) − (1, 3) = [0, 1]
+</section>
```
````
<section id='description'>
All real numbers form the uncountable set ℝ.  Among its subsets, relatively simple are the convex sets, each expressed as a range between two real numbers <i>a</i> and <i>b</i> where <i>a</i> ≤ <i>b</i>.  There are actually four cases for the meaning of "between", depending on open or closed boundary:

<ul>
  <li>[<i>a</i>, <i>b</i>]: {<i>x</i> | <i>a</i> ≤ <i>x</i> and <i>x</i> ≤ <i>b</i> }</li>
  <li>(<i>a</i>, <i>b</i>): {<i>x</i> | <i>a</i> < <i>x</i> and <i>x</i> < <i>b</i> }</li>
  <li>[<i>a</i>, <i>b</i>): {<i>x</i> | <i>a</i> ≤ <i>x</i> and <i>x</i> < <i>b</i> }</li>
  <li>(<i>a</i>, <i>b</i>]: {<i>x</i> | <i>a</i> < <i>x</i> and <i>x</i> ≤ <i>b</i> }</li>
</ul>

Note that if <i>a</i> = <i>b</i>, of the four only [<i>a</i>, <i>a</i>] would be non-empty.

<strong>Task</strong>

<ul>
  <li>Devise a way to represent any set of real numbers, for the definition of "any" in the implementation notes below.</li>
  <li>Provide methods for these common set operations (<i>x</i> is a real number; <i>A</i> and <i>B</i> are sets):</li>
  <ul>
    <li>
      <i>x</i> ∈ <i>A</i>: determine if <i>x</i> is an element of <i>A</i><br>
      example: 1 is in [1, 2), while 2, 3, ... are not.
    </li>
    <li>
      <i>A</i> ∪ <i>B</i>: union of <i>A</i> and <i>B</i>, i.e. {<i>x</i> | <i>x</i> ∈ <i>A</i> or <i>x</i> ∈ <i>B</i>}<br>
      example: [0, 2) ∪ (1, 3) = [0, 3); [0, 1) ∪ (2, 3] = well, [0, 1) ∪ (2, 3]
    </li>
    <li>
      <i>A</i> ∩ <i>B</i>: intersection of <i>A</i> and <i>B</i>, i.e. {<i>x</i> | <i>x</i> ∈ <i>A</i> and <i>x</i> ∈ <i>B</i>}<br>
      example: [0, 2) ∩ (1, 3) = (1, 2); [0, 1) ∩ (2, 3] = empty set
    </li>
    <li>
      <i>A</i> - <i>B</i>: difference between <i>A</i> and <i>B</i>, also written as <i>A</i> \ <i>B</i>, i.e. {<i>x</i> | <i>x</i> ∈ <i>A</i> and <i>x</i> ∉ <i>B</i>}<br>
      example: [0, 2) − (1, 3) = [0, 1]
    </li>
  </ul>
</ul>
</section>
````

scissorsneedfoodtoo commented on curriculum/challenges/english/08-coding-interview-prep/rosetta-code/set-of-real-numbers.english.md:
```diff
@@ -0,0 +1,161 @@
+
+## Instructions
+
+<section id='instructions'>
+Write a function that takes 2 objects, a string and an array as parameters. The objects represents the set and have attributes: low, high and rangeType. The rangeType can have values 0, 1, 2 and 3 for CLOSED, BOTH_OPEN, LEFT_OPEN and RIGHT_OPEN respectively. The function should implement a set using this information. The string represents the operation to be performed on the sets. It can be: "uni ...
+</section>
```
````
<section id='instructions'>

Write a function that takes 2 objects, a string and an array as parameters. The objects represents the set and have attributes: `low`, `high` and `rangeType`. 

The `rangeType` can have values 0, 1, 2 and 3 for `CLOSED`, `BOTH_OPEN`, `LEFT_OPEN` and `RIGHT_OPEN`, respectively. The function should implement a set using this information. 

The string represents the operation to be performed on the sets. It can be: `"union"`, `"intersect"` and `"subtract"` (difference). 

After performing the operation, the function should check if the values in the array are present in the resultant set and store a corresponding boolean value to an array. The function should return this array.

</section>
````

scissorsneedfoodtoo commented on curriculum/challenges/english/08-coding-interview-prep/rosetta-code/sha-1.english.md:
```diff
@@ -0,0 +1,167 @@
+SHA-1 often appears in security protocols; for example, many HTTPS websites use RSA with SHA-1 to secure their connections.
+BitTorrent uses SHA-1 to verify downloads.
+Git and Mercurial use SHA-1 digests to identify commits.
+A US government standard, <a href="http://rosettacode.org/wiki/SHA-1/FIPS-180-1" target="_blank">FIPS 180-1</a>, defines SHA-1.
```
Spacing here seems alright. Let's just go to the secure version of the page:

```
A US government standard, <a href="https://rosettacode.org/wiki/SHA-1/FIPS-180-1" target="_blank">FIPS 180-1</a>, defines SHA-1.
```

scissorsneedfoodtoo commented on curriculum/challenges/english/08-coding-interview-prep/rosetta-code/sha-256.english.md:
```diff
@@ -0,0 +1,411 @@
+## Description
+
+<section id='description'>
+<b><a href="https://en.wikipedia.org/wiki/SHA-256" target="_blank">SHA-256</a></b> is the recommended stronger alternative to <a href="http://rosettacode.org/wiki/SHA-1" target="_blank">SHA-1</a>.
```
```
<b><a href="https://en.wikipedia.org/wiki/SHA-256" target="_blank">SHA-256</a></b> is the recommended stronger alternative to <a href="https://rosettacode.org/wiki/SHA-1" target="_blank">SHA-1</a>.
```

scissorsneedfoodtoo commented on curriculum/challenges/english/08-coding-interview-prep/rosetta-code/sha-256.english.md:
```diff
@@ -0,0 +1,411 @@
+## Instructions
+
+<section id='instructions'>
+Write a function that takes a string as a parameter and returns its the SHA-256 digest.
```
Write a function that takes a string as a parameter and returns its SHA-256 digest.

scissorsneedfoodtoo commented:
@moT01, awesome, thank you for creating Guide articles for all of those challenges.

@NitronR, I noticed a few things and will leave some review comments.

scissorsneedfoodtoo commented:
@thecodingaviator, could you take a look at this when you get a chance?

thecodingaviator commented:
@NitronR drop a reply in after you've made the above changes, and I'll have a look!

NitronR commented:
@scissorsneedfoodtoo Thanks for the review. I have done the changes.
@thecodingaviator Can you please take a look?

thecodingaviator commented on curriculum/challenges/english/08-coding-interview-prep/rosetta-code/self-referential-sequence.english.md:
```diff
@@ -0,0 +1,151 @@
+## Description
+
+<section id='description'>
+There are several ways to generate a self-referential sequence. One very common one (the <a href="https://rosettacode.org/wiki/Look-and-say sequence" target="_blank">Look-and-say sequence</a>) is to start with a positive integer, then generate the next term by concatenating enumerated groups of adjacent alike digits:
```
We might want to add what a self-referential sequence is

NitronR replied:
I did not find a formal definition but mostly its just the essence of the words "self-referential sequence" which we can describe. We can add this line: A self-referential sequence is a sequence whose elements are generated by using its current elements.
What do you think?

thecodingaviator replied:
I guess rather than coming up with one of our own, we should leave it to the user to understand in that case?

scissorsneedfoodtoo replied:
@thecodingaviator, I think it's okay to let learners look up the definition or go to the Look-and-say sequence to get a better understanding of what a self-referential sequence is. Also if they read on they get some concrete examples.

thecodingaviator commented on curriculum/challenges/english/08-coding-interview-prep/rosetta-code/self-referential-sequence.english.md:
```diff
@@ -0,0 +1,151 @@
+
+Sort the digits largest to smallest. Do not include counts of digits that do not appear in the previous term.
+
+Depending on the seed value, series generated this way always either converge to a stable value or to a short cyclical pattern. (For our purposes, I'll use converge to mean an element matches a previously seen element.) The sequence shown, with a seed value of 0, converges to a stable value of 1433223110 after 11 iterations. The seed value that converges most quickly is 22. It goes stable after t ...
```
```suggestion
Depending on the seed value, series generated this way always either converge to a stable value or to a short cyclical pattern. (For our purposes, I'll use converge to mean an element matches a previously seen element.) The sequence shown, with a seed value of 0, converges to a stable value of 1433223110 after 11 iterations. The seed value that converges most quickly is 22. It goes stable after the first element. (The next element is 22, which has been seen before.)
```
Uses self-referencing language `I'll` and such. Please revise this

scissorsneedfoodtoo replied:
This is a tricky once since it's in the original Rosetta Code description. Generally we want to stick closely to the original description, but here it does make sense to modify it.

How about we change the sentence slightly to remove the word "I'll":

"(For our purposes, converge meaning when an element matches a previously seen element.)"

thecodingaviator replied:
That sounds great!

thecodingaviator commented on curriculum/challenges/english/08-coding-interview-prep/rosetta-code/semiprime.english.md:
```diff
@@ -0,0 +1,116 @@
+## Instructions
+
+<section id='instructions'>
+Write a function that returns true if a number is semiprime, or false if it is not.
```
```suggestion
Write a function that returns true if a number is semiprime, and false if it is not.
```

scissorsneedfoodtoo replied:
This could go either way. It seems like our current challenge use both "and" or "or" for this type of sentence.

thecodingaviator commented on curriculum/challenges/english/08-coding-interview-prep/rosetta-code/sha-1.english.md:
```diff
@@ -0,0 +1,167 @@
+forumTopicId: 385326
+---
+
+## Description
+
+<section id='description'>
+<b>SHA-1</b> or <b>SHA1</b> is a one-way hash function; it computes a 160-bit message digest.
+SHA-1 often appears in security protocols; for example, many HTTPS websites use RSA with SHA-1 to secure their connections.
+BitTorrent uses SHA-1 to verify downloads.
+Git and Mercurial use SHA-1 digests to identify commits.
+A US government standard, <a href="https://rosettacode.org/wiki/SHA-1/FIPS-180-1" target="_blank">FIPS 180-1</a>, defines SHA-1.
+</section>
+
+## Instructions
+
+<section id='instructions'>
+Write a function that returns the SHA-1 message digest for a given string.
+</section>
```
I think we should add how SHA-1 is generated and such relevant information.

scissorsneedfoodtoo replied:
I don't think this is necessary in this case. If we do, the description will be quite long.

The included link toward the bottom of the description breaks things down well. Alternatively, we could also include a link to Wikipedia at the top:

`<a href="https://en.wikipedia.org/wiki/SHA-1" target="_blank"><b>SHA-1</b></a> or <b>SHA1</b> is a one-way hash function; it computes a 160-bit message digest.`

thecodingaviator replied:
Didn't see the link before. That works!

NitronR commented:
@scissorsneedfoodtoo @thecodingaviator Thanks for helping out with this!

moT01 reviewed (changes requested):
Can we also add the new `isHidden: false` flag at the top of these.

moT01 commented:
I have blocked this for now. See the [pinned code freeze issue](https://github.com/freeCodeCamp/freeCodeCamp/issues/38308)

raisedadead commented:
Limitations removed on Coding Interview Prep Challenges.

scissorsneedfoodtoo commented:
@NitronR, hope you don't mind, but I went in and added the `isHidden` flags for you.

Thanks again for all of your hard work porting these challenges over. Looking forward to your next PR :+1: :+1:

NitronR commented:
@scissorsneedfoodtoo Thank you for the help!

moT01 reviewed (changes requested):
I made a few suggestions @NitronR. Nice job on these 👍 Could you complete the checklist in the PR template please?

NitronR commented on curriculum/challenges/english/10-coding-interview-prep/rosetta-code/self-referential-sequence.english.md:
```diff
@@ -0,0 +1,152 @@
+## Instructions
+
+<section id='instructions'>
+Write a function that takes the seed value as parameter and returns the self referential sequence as an array.
```
@moT01 I think the second method is the one in which the series converges and hence it is the focus of the task. For the first method, it is stated: "The terms generated grow in length geometrically and never converge". As it is an infinite sequence, it cannot have a finite answer. I also think that we can make the instructions more clear.
```
Write a function that takes the seed value as parameter, generate a  self referential sequence until it converges and return it as an array.
```
This would imply to use the second method as the first one do not converge.

scissorsneedfoodtoo replied:
@NitronR, thanks for improving these instructions. Your proposal sounds much clearer. I just want to suggest a couple of small changes:

`Write a function that takes the seed value as parameter, generates a self referential sequence until it converges, and returns it as an array.`

NitronR commented on curriculum/challenges/english/10-coding-interview-prep/rosetta-code/self-referential-sequence.english.md:
```diff
@@ -0,0 +1,152 @@
+    testString: assert(typeof selfReferential === 'function');
+  - text: <code>selfReferential(40)</code> should return a array.
+    testString: assert(Array.isArray(selfReferential(40)));
+  - text: <code>selfReferential(40)</code> should return <code>["1410", "142110", "14123110", "1413124110", "2413125110", "151413224110", "152413225110", "251413324110", "152423224110", "152413423110"]</code>.
```
Makes sense. I will update the test cases and the solution.

NitronR committed:
fix(interview-prep): Fixed descriptions and test cases

NitronR commented:
@moT01 @scissorsneedfoodtoo I have made the changes. Can you please check?

moT01 reviewed (approved):
LGTM @NitronR 🎉 Thanks for all the work on this 🎉

scissorsneedfoodtoo commented:
Thanks for all your hard work on these ports @NitronR and @moT01 for your review. Everything LGTM :+1: :+1:
